### PR TITLE
[feat] Added Kotlinx Serialization support

### PR DESCRIPTION
**Pull request by ctasada**
Added basic support for KotlinX Serialization classes.

This new module serializes Kotlin `@Serializable` classes respecting the `@SerialName` annotation and types.

Solves https://github.com/springwolf/springwolf-core/issues/589

### CODE REVIEW

**Review by timonback — COMMENTED** [review]
I like this implementation as it is a lot more compact and an add-on module seems appropriate.

Can you add a (small) test case for useFqn=true (the name field is sufficient)?

**Comment by ctasada** [comment]
@timonback done.

I will keep iterating over next PRs, but I think it's good enough to go, at least as a `beta`

File: springwolf-add-ons/springwolf-kotlinx-serialization-model-converter/src/main/kotlin/io/github/springwolf/addons/kotlinx_serialization_model_converter/converter/KotlinxSerializationModelConverter.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            propertySchema = ArraySchema()
+            val value = (property.returnType.javaType as ParameterizedType).actualTypeArguments[0]
+            propertySchema.items = resolveRefSchema(value, context)
+            propertySchema.uniqueItems = true
+        } else if (property.returnType.jvmErasure.java.isAssignableFrom(Map::class.java)) {
+            val mapType = (property.returnType.javaType as ParameterizedType)
+            val value = mapType.actualTypeArguments[1]
+            propertySchema = MapSchema().additionalProperties(context.resolve(AnnotatedType(value)))
+        } else if (property.returnType.jvmErasure.java.isAssignableFrom(Byte::class.java)) {
+            propertySchema = IntegerSchema()
```
Comment (sam0r040) [inline springwolf-add-ons/springwolf-kotlinx-serialization-model-converter/src/main/kotlin/io/github/springwolf/addons/kotlinx_serialization_model_converter/converter/KotlinxSerializationModelConverter.kt:82]: Why do we need to convert Byte to Integerschema? https://www.asyncapi.com/docs/reference/specification/v3.0.0#dataTypeFormat looks like StringSchema with format byte would be the correct schema type and its also what would be generated by default...

**Comment by sam0r040** [comment]
Hi @ctasada,

we had a look at the changes and like it a lot more than the one that you have shown us before. We like to see:

- More simplified code by using more Kotlin features unless we missed something
- More fine grained and isolated test cases that make understanding and debugging easier

We created https://github.com/ctasada/springwolf-core/pull/3 as as suggestion for both and postpone the release till Wednesday (27.3.)  to give you some time to review the suggestion. Tests can be refactored later, but we would like to make the production code simpler.

From our side it would be sufficient to just merge the PR https://github.com/ctasada/springwolf-core/pull/3 and then we can merge this PR without the need for further changes for the march release.

File: springwolf-add-ons/springwolf-kotlinx-serialization-model-converter/src/main/kotlin/io/github/springwolf/addons/kotlinx_serialization_model_converter/converter/KotlinxSerializationModelConverter.kt (hunk 1 above)
Comment (ctasada) [reply springwolf-add-ons/springwolf-kotlinx-serialization-model-converter/src/main/kotlin/io/github/springwolf/addons/kotlinx_serialization_model_converter/converter/KotlinxSerializationModelConverter.kt]: @sam0r040 good catch. It's now fixed

**Comment by ctasada** [comment]
@sam0r040 @timonback I merged the suggested improvements, fixed the `byte` bug and refactor the tests a bit.

Hopefully we are good to go

**Review by timonback — APPROVED** [review]
Looks great @ctasada and looking forward to the next improvement to move it beyond the beta status.

**Pull request merged**